### ISSUE TITLE
named_tag attribute for ecx_l2_connection is not idempotent

### ISSUE THREAD

**Issue by wrgeorge1983** [equinix/terraform-provider-equinix#97]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Terraform Version

Terraform v1.0.11
on linux_amd64
+ provider registry.terraform.io/equinix/equinix v1.2.0
+ provider registry.terraform.io/hashicorp/azurerm v2.95.0
+ provider registry.terraform.io/hashicorp/random v3.1.0

### Affected Resource(s)

ecx_l2_connection 

### Terraform Configuration Files

```tf

data "equinix_ecx_port" "this" {
  name = "this_port"
}

resource "equinix_ecx_l2_connection" "ecx_connection" {

  name                  = "this_pri"
  speed                 = 50
  speed_unit            = "MB"
  notifications         = ["foo@bar.com"]
  purchase_order_number = "this"

  port_uuid         = data.equinix_ecx_port.this.id
  vlan_stag         = 165

  secondary_connection {
    name = "this_sec"
    port_uuid = data.equinix_ecx_port.this.id
    vlan_stag = 166
  }

  seller_metro_code = "DC"
  seller_region     = "eastus"
  profile_uuid      = data.equinix_ecx_l2_sellerprofile.this.id
  authorization_key = var.auth_key
  named_tag = "Private"
}
```

### Expected Behavior

Applying the above TF code should succeed in creating the resources, and correctly identify them as unchanged on future runs

### Actual Behavior

The initial creation is a success, but future runs detect a "change" on the named_tag attribute ("PRIVATE" vs "Private").

```
  # module.ecx.equinix_ecx_l2_connection.ecx_connection must be replaced
-/+ resource "equinix_ecx_l2_connection" "ecx_connection" {
      - device_uuid           = "" -> null
      ~ id                    = "c12cc978-...-de255a524478" -> (known after apply)
        name                  = "buildTheFactory-dev_pri"
      ~ named_tag             = "PRIVATE" -> "Private" # forces replacement
      ~ provider_status       = "PENDING_BGP" -> (known after apply)
      ~ redundancy_type       = "PRIMARY" -> (known after apply)
      ~ redundant_uuid        = "" -> (known after apply)
      ~ status                = "PENDING_BGP_PEERING" -> (known after apply)
      ~ uuid                  = "c12cc978-...-de255a524478" -> (known after apply)
      - vlan_ctag             = 0 -> null
      ~ zside_port_uuid       = "66284add-...-30ac094f8af1" -> (known after apply)
      ~ zside_vlan_ctag       = 0 -> (known after apply)
      ~ zside_vlan_stag       = 79 -> (known after apply)
        # (10 unchanged attributes hidden)

      ~ secondary_connection {
          + authorization_key   = (known after apply)
          ~ device_interface_id = 0 -> (known after apply)
            name                = "buildTheFactory-dev_sec"
          + profile_uuid        = (known after apply)
          + provider_status     = (known after apply)
          + redundancy_type     = (known after apply)
          + redundant_uuid      = (known after apply)
          + seller_metro_code   = (known after apply)
          + seller_region       = (known after apply)
          ~ speed               = 0 -> (known after apply)
          + speed_unit          = (known after apply)
          + status              = (known after apply)
          + uuid                = (known after apply)
          - vlan_ctag           = 0 -> null
          + zside_port_uuid     = (known after apply)
          ~ zside_vlan_ctag     = 0 -> (known after apply)
          ~ zside_vlan_stag     = 0 -> (known after apply)
            # (2 unchanged attributes hidden)
        }
    }

Plan: 1 to add, 0 to change, 1 to destroy.
```

Also attempting to use "PRIVATE" as the named_tag as a workaround fails:
```
╷
│ Error: expected named_tag to be one of [Private Public Microsoft Manual], got PRIVATE
│ 
│   with module.ecx.equinix_ecx_l2_connection.ecx_connection,
│   on ../../modules/ecx_vc_red/fp.tf line 31, in resource "equinix_ecx_l2_connection" "ecx_connection":
│   31:   named_tag = "PRIVATE"
│ 
╵
```

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by wrgeorge1983** [comment]
If possible it'd probably be best if either the TF Provider or ECX API just normalized case to whatever it expects/needs.

**Comment by ocobleseqx** [comment]
> If possible it'd probably be best if either the TF Provider or ECX API just normalized case to whatever it expects/needs.

Hi @wrgeorge1983 We will keep the current behavior in terraform as there is no restriction in the API. To prevent the plan from reflecting a change, we will suppress the capitalization difference check for this field

This issue will be addressed together with #98
